includes/Checker/AJAX_Runner.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\AJAX_Runner
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+use WordPress\Plugin_Check\Utilities\Plugin_Request_Utility;
+
+/**
+ * AJAX Runner class.
+ *
+ * @since n.e.x.t
+ */
+class AJAX_Runner extends Abstract_Check_Runner {
+
+	/**
+	 * An instances of the Checks class.
+	 *
+	 * @since n.e.x.t
+	 * @var Checks
+	 */
+	protected $checks;
+
+	/**
+	 * Checks if the current request is an AJAX request for the Plugin Checker.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool
+	 */
+	public function is_plugin_check() {
+		if ( 0 !== strpos( $_SERVER['REQUEST_URI'], '/admin-ajax.php' ) ) {
+			return false;
+		}
+
+		if ( ! isset( $_REQUEST['action'] ) || 'plugin_check_run_checks' !== $_REQUEST['action'] ) {
+			return false;
+		}
+
+		if ( ! check_ajax_referer( 'plugin_check_run_checks' ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retruns an instance of the Checks class.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return Checks
+	 *
+	 * @throws Exception Thrown if the plugin main file cannot be found based on the AJAX input.
+	 */
+	protected function get_checks_instance() {
+		if ( ! isset( $this->checks ) ) {
+			// Get the plugin name from the AJAX request.
+			$plugin_slug = isset( $_REQUEST['plugin'] ) ? $_REQUEST['plugin'] : '';
```

Comment:
Let's be strict. If this is empty or not set, this should probably throw an exception.

Letting bad requests silently go through makes debugging problems unnecessarily hard later.

Reply:
Thanks @felixarntz the `Plugin_Request_Utility::get_plugin_basename_from_input` call does throw an exception when the plugin is empty/invalid. But I've added a new exception in this method to capture this earlier.

Reply:
@jjgrainger Ah good point, I had missed that. I guess it's okay either way then 👍